src/Grb.Building.Api/Infrastructure/Startup.cs

```diff
@@ -0,0 +1,204 @@
+    using Be.Vlaanderen.Basisregisters.Api;
+    using Be.Vlaanderen.Basisregisters.DataDog.Tracing.Microsoft;
+    using Configuration;
+    using IdentityModel.AspNetCore.OAuth2Introspection;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.Mvc.ApiExplorer;
+    using Microsoft.AspNetCore.Mvc.Infrastructure;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Diagnostics.HealthChecks;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.OpenApi.Models;
+    using Modules;
+    using Options;
+
+    /// <summary>Represents the startup process for the application.</summary>
+    public class Startup
+    {
+        private const string DatabaseTag = "db";
+
+        private IContainer _applicationContainer;
+
+        private readonly IConfiguration _configuration;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public Startup(
+            IConfiguration configuration,
+            ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _loggerFactory = loggerFactory;
+        }
+
+        /// <summary>Configures services for the application.</summary>
+        /// <param name="services">The collection of services to configure the application with.</param>
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            var oAuth2IntrospectionOptions = _configuration
+                .GetSection(nameof(OAuth2IntrospectionOptions))
+                .Get<OAuth2IntrospectionOptions>();
+
+            var baseUrl = _configuration.GetValue<string>("BaseUrl");
+            var baseUrlForExceptions = baseUrl.EndsWith("/")
+                ? baseUrl.Substring(0, baseUrl.Length - 1)
+                : baseUrl;
+
+            services.AddAcmIdmAuthentication(oAuth2IntrospectionOptions!);
+            services
+                .ConfigureDefaultForApi<Startup>(new StartupConfigureOptions
+                    {
+                        Cors =
+                        {
+                            Origins = _configuration
+                                .GetSection("Cors")
+                                .GetChildren()
+                                .Select(c => c.Value)
+                                .ToArray()
+                        },
+                        Server =
+                        {
+                            BaseUrl = baseUrlForExceptions
+                        },
+                        Swagger =
+                        {
+                            ApiInfo = (provider, description) => new OpenApiInfo
+                            {
+                                Version = description.ApiVersion.ToString(),
+                                Title = "Basisregisters Vlaanderen Building Registry GRB API",
```

Comment:
Title should be Basisregisters GRB API?

Reply:
Don't think so. Parcel will probably also have its API.